### PR TITLE
fix: improve error message for status db

### PR DESCRIPTION
**Pull request by rgrinberg**
When we tell the user a stanza is forbidden under [(include_subdirs
$type)] we should write the correct $type rather than guess it's
[unqualified]

Adding Ali as a reviewer as this only affects error messages for coq

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

**Comment by rgrinberg** [comment]
Coq has the qualified option.

Moreover, I'm going to lift the requirement for OCaml as well.

**Comment by Alizter** [comment]
@rgrinberg In that case looks fine. I guess you can't write a test to for the behaviour yet without coq lang right? If you lift the restriction on OCaml we should remember to add `include-subdirs/error4.t` or something.

File: src/dune_rules/dir_status.ml [hunk 1]
```diff
@@ -107,13 +113,22 @@ end = struct
...
-              check_no_module_consumer d.stanzas;
+              let+ () =
+                match find_module_stanza d.stanzas with
+                | None -> Memo.return ()
+                | Some loc -> (
+                  let+ group = get ~dir:group_root in
+                  match group with
+                  | Group_root (_, (_, qualification), _) ->
+                    error_no_module_consumer ~loc qualification
+                  | _ -> assert false)
```
Comment (Alizter) [inline src/dune_rules/dir_status.ml:128]: nit: code error here instead?

**Review by Alizter — APPROVED** [review]

**Comment by rgrinberg** [comment]
It's just an error message anyway. There's more important things to test.

**Pull request merged**